src/matrices.jl

```diff
@@ -0,0 +1,75 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    JMatrix{T}
+
+Square matrix of ones.
+"""
+struct JMatrix{T} end
+(J::JMatrix{T})(d::Integer) where {T} = Ones{T}(d, d)
+(*)(::JMatrix, v::AbstractVector) = Fill(sum(v), length(v))
```

Comment:
Let's remove the parenthesis before and after * as well in the next round of changes.

Reply:
done